### PR TITLE
feat(frontend): App set tab

### PR DESCRIPTION
**Pull request by fatonramadani**
Screenshot 2023-03-20 at 11 31 21

### CODE REVIEW

File: frontend/src/lib/components/apps/components/helpers/RunnableComponent.svelte [hunk 1]
```diff
@@ -221,7 +223,12 @@
 	}
 
 	export async function runComponent() {
-		await executeComponent()
+		try {
+			await executeComponent()
+			dispatch('success')
+		} catch (e) {
+			console.error(e)
+		}
```
Comment (rubenfiszel) [inline frontend/src/lib/components/apps/components/helpers/RunnableComponent.svelte]: remove that code and look at setResult instead. The dispatch would be after checking result.error

File: frontend/src/lib/components/apps/editor/component/components.ts [hunk 2]
```diff
@@ -340,6 +338,12 @@ export const components = {
 					value: false,
 					fieldType: 'boolean',
 					onlyStatic: true
+				},
+				setTab: {
+					type: 'static',
+					value: { id: '', index: 0 },
```
Comment (rubenfiszel) [inline frontend/src/lib/components/apps/editor/component/components.ts]: should be an empty array

File: frontend/src/lib/components/apps/editor/component/components.ts [hunk 3]
```diff
@@ -362,6 +366,12 @@ export const components = {
 			},
 			recomputeIds: undefined,
 			configuration: {
+				setTab: {
+					type: 'static',
+					value: { id: '', index: 0 },
```
Comment (rubenfiszel) [inline frontend/src/lib/components/apps/editor/component/components.ts]: should be an empty array

File: frontend/src/lib/components/apps/inputType.ts [hunk 4]
```diff
@@ -163,6 +164,7 @@ export type AppInput =
 	| (AppInputSpec<'array', string[], 'select'> & StaticOptions)
 	| AppInputSpec<'array', object[], 'labeledresource'>
 	| AppInputSpec<'labeledresource', object>
+	| AppInputSpec<'tab-select', object>
```
Comment (rubenfiszel) [inline frontend/src/lib/components/apps/inputType.ts]: Look at labelresource to see how to do an array of tab-select

File: frontend/src/lib/components/apps/components/helpers/RunnableComponent.svelte (hunk 1 above)
Comment (fatonramadani) [reply frontend/src/lib/components/apps/components/helpers/RunnableComponent.svelte]: fixed!

File: frontend/src/lib/components/apps/editor/component/components.ts (hunk 2 above)
Comment (fatonramadani) [reply frontend/src/lib/components/apps/editor/component/components.ts]: Done

File: frontend/src/lib/components/apps/components/helpers/RunnableComponent.svelte [hunk 5]
```diff
@@ -255,6 +261,9 @@
 		if (res?.error) {
 			recordError(res.error)
 		}
+
```
Comment (rubenfiszel) [inline frontend/src/lib/components/apps/components/helpers/RunnableComponent.svelte:263]: Put it inside else :)

File: frontend/src/lib/components/apps/components/helpers/RunnableWrapper.svelte [hunk 6]
```diff
@@ -64,6 +66,14 @@
 		wrapperClass={runnableClass}
 		wrapperStyle={runnableStyle}
 		{render}
+		on:success={() => {
+			if (Array.isArray(setTab)) {
+				setTab.forEach((tab) => {
+					const { id, index } = tab
+					$componentControl[id].setTab?.(index)
+				})
+			}
```
Comment (rubenfiszel) [inline frontend/src/lib/components/apps/components/helpers/RunnableWrapper.svelte]: let's handle the goto and gotoNewTab here

Comment (fatonramadani) [reply frontend/src/lib/components/apps/components/helpers/RunnableWrapper.svelte]: Done! I've also moved the recompute code here as well

**Pull request merged**